src/client/src/components/chatpage.js

```diff
@@ -32,58 +28,61 @@ class ChatPage extends React.Component {
             userRooms: []
         }
 
-        this.handleChange = this.handleChange.bind(this)
+        this.handleUserInput = this.handleUserInput.bind(this)
         this.sendMessage = this.sendMessage.bind(this)
     }
     componentDidMount(){
-        socket.on(protocols.CONNECTION, _ => this.setState({connected: true}))
-        socket.on(protocols.MESSAGE, data => this.populateMessages(data))
-        socket.on(protocols.JOIN, data => this.populateData(data))
+        socket.on(protocols.CONNECTION, () => this.setState({connected: true}))
+        socket.on(protocols.MESSAGE, data => this.storeMessage(data))
+        socket.on(protocols.JOIN, data => this.updateJoinedRooms(data))
         socket.emit(protocols.JOIN, {roomId: 'df7a1d12-7be0-4aab-8685-0cbf237bb135'}) // Development purposes - remove in production
     }
-    populateData(data){
-        const {roomId, username} = data
-        this.setState(updateState('userRooms', this.state.userRooms, {roomId, username}))
+    updateJoinedRooms(data){
+        const {roomId} = data
+        const usernames = data.username
+        this.setState({userRooms: this.state.userRooms.concat({roomId, usernames})})
     }
     changeSelectedRoom(roomDetails){
         const {roomId} = roomDetails
         this.setState({selectedRoom: roomId}) 
     }
-    populateMessages(messageData){
+    storeMessage(messageData){
         const {roomId, username, message, timestamp} = messageData
-        //Create a clone to mutate
-        const messages = Object.assign({}, this.state.messages)
-        //Add new message to the list
-        messages[roomId] = (messages[roomId] || []).concat({username, message, timestamp})
-        //Update list
+        const messages = _.clone(this.state.messages)
+
+        if(!_.has(messages, roomId)) messages[roomId] = []
+        messages[roomId].push({username, message, timestamp})
+
         this.setState({messages})
     }
     findUsersOfRoom(roomId){
-        const value = roomId ? this.state.userRooms.find(obj => obj.roomId === roomId) : {}
-        return value.username || ''
+        const roomObject = this.state.userRooms.find(room => room.roomId === roomId)
+        return _.get(roomObject, 'usernames', '')
     }
     generateMessages(){
-        //Check that a room is selected
         if(this.state.selectedRoom){
-                //Check if the room has messages
             if(!this.state.messages[this.state.selectedRoom]) return
-
+        
             return this.state.messages[this.state.selectedRoom]
-                .map((msg, i) => <li key={i}>{`${msg.username}:\t ${msg.message} \t ${msg.timestamp}`}</li>)
+                .map((msg, i) => 
+                    <li key={i}>
+                        {`${msg.username}:\t ${msg.message} \t ${msg.timestamp}`}
+                    </li>
+                )
         }
-
         return <h6>Please join a room before attempting to load messages</h6>
     }
-    handleChange(e) {
+    handleUserInput(e) {
```

Comment:
`e` -> `event`
Avoid abbreviations, they might be obvious for the writer, but not for the reader, who reads the function out of context at first. Function should be self-explainable.